internal/cmd/stack/stack_selector.go

```diff
@@ -31,14 +34,28 @@ func getStackID(cliCtx *cli.Context) (string, error) {
 func getStack(cliCtx *cli.Context) (*stack, error) {
 	if cliCtx.IsSet(flagStackID.Name) {
 		stackID := cliCtx.String(flagStackID.Name)
-		stack, err := stackGetByID(cliCtx.Context, stackID)
+		stack, found, err := stackGetByID(cliCtx.Context, stackID)
 		if err != nil {
 			return nil, fmt.Errorf("failed to check if stack exists: %w", err)
 		}
-		if errors.Is(err, errNoStackFound) {
```

Comment:
It doesn't work, replaced it with a bool variable

Reply:
We can't reach this condition with a non-nil error because we handle all errors on line 38.

We have to reorder the conditions to make it work, like this:

```
stack, err := stackGetByID(cliCtx.Context, stackID)
if errors.Is(err, errNoStackFound) {
	return nil, fmt.Errorf("stack with id %q could not be found. Please check that the stack exists and that you have access to it. To list available stacks run: spacectl stack list", stackID)
}
if err != nil {
	return nil, fmt.Errorf("failed to check if stack exists: %w", err)
}
```

Reply:
True, I didn't spot that. Thanks!

Reply:
def7656745a4e1ffb4c1becb5ee18c02fb95ecee